packages/gatsby-remark-embed-snippet/README.md

```diff
@@ -110,6 +110,30 @@ baz: "highlighted"
 quz: "highlighted"
 ```
 
+### Hide Lines
+
+Also is possible specify a range of lines to be hide.
+
+#### JavaScript example
+
+```js
+// hideline-range{1-3,9-11}
```

Comment:
I think we should have an aligned standard between hideline and highlight-line. That said, I'm not sure if auto-trimming blank lines is a good idea. For me it would add more confusion (since it's already complicated enough). I'm thinking maybe we could hold this off and work out a first version of it. :)

Reply:
I'm not suggesting we remove all blank lines, but that we "trim" leading and trailing black lines- because you would never want them, I think.

Reply:
Ah that makes more sense to me now. I agree with trimming the leading and trailing blank lines. Good idea.

Reply:
> "I wonder if our example should show the `hideline-range` tags right above the hidden ranges (which seems to be easier to maintain, in  practice)"

I do not know if I understood correctly, but I think it would be more interesting to keep a standard on the use of the directives.
It might be confusing to have this behavior only for the `hideline-range` directive, since the ` highlight-range` directive does not have this behavior even for large files.

We could generate a first version only with existing behavior on the PR, and then we can think better and make some improvement 🙂. What do you think ?

Reply:
About auto-trim, I got it better now and it seems like a good idea. 😃

Reply:
> I also think we should auto-trim so the blank lines don't need to be explicitly hidden.

@bvaughn If I understood the code well, the hidden lines are already being hidden automatically. 
Is that what you meant?

[gatsby-remark-embed-snippet/src/index.js#L164](https://github.com/washingtonsoares/gatsby/blob/6b37cdf287eba43788b080d1bffcd8a8bc4e1eee/packages/gatsby-remark-embed-snippet/src/index.js#L164)

cc @cyan33

Reply:
Sorry, I meant leading and trailing lines should be auto-hidden.

For example, given the original example code in this thread:
```js
// hideline-range{1-2}
import React from "react"
import ReactDOM from "react-dom"

function App() {
  return (
    <h1>Hello world!</h1>
  )
}

// hideline-range{1-2}
const rootElement = document.getElementById("root")
ReactDOM.render(<App />, rootElement)
```

Without trimming, we would end up with:
```

function App() {
  return (
    <h1>Hello world!</h1>
  )
}

```

(Note the empty lines above and below.)

Reply:
@bvaughn I wrote a test case to reproduce this: [src/\_\_tests__/index.js#L440](https://github.com/washingtonsoares/gatsby/blob/5ecf804d6de3b93a25bf25541f851a89e3540bf3/packages/gatsby-remark-embed-snippet/src/__tests__/index.js#L440)

And the output is:

```javascript
ReactDOM.render(
    <div>
    <ul>
        <li>Not hidden</li>
        <li>Not hidden</li>
        <li>Not hidden</li>
    </ul>
    </div>,
    document.getElementById('root')
);
```

(without leading and trailing lines)

Is that what you meant?

Reply:
Looks like it

Reply:
@bvaughn so, do you think we're ready to merge the PR? :smile:

Reply:
I don't have the bandwidth to review or merge anything Gatsby related right now. I'm sorry man. I'm swamped with React stuff.

Reply:
Ok no problem. Thanks for your help.

Reply:
@m-allanson @pieh  @KyleAMathews what do you guys think?